packages/ember-metal/lib/property_set.js

```diff
@@ -39,20 +39,21 @@ export function set(obj, keyName, value, tolerant) {
   assert(`The key provided to set must be a string, you passed ${keyName}`, typeof keyName === 'string');
   assert(`'this' in paths is not supported`, !pathHasThis(keyName));
 
-  var meta, possibleDesc, desc;
+  let meta, desc;
   if (obj) {
     meta = peekMeta(obj);
-    possibleDesc = obj[keyName];
-    desc = (possibleDesc !== null && typeof possibleDesc === 'object' && possibleDesc.isDescriptor) ? possibleDesc : undefined;
+    desc = meta && meta.peekDescs(keyName);
   }
 
-  var isUnknown, currentValue;
+  let isUnknown, currentValue;
   if (desc === undefined && isPath(keyName)) {
     return setPath(obj, keyName, value, tolerant);
   }
 
   assert('calling set on destroyed object', !obj.isDestroyed);
 
+
+  // soon this will be just obj[keyName] = value, as it shouldn be...
```

Comment:
this will  depend on the outcome of https://github.com/emberjs/ember.js/pull/12852/files#r50556996